### DISCUSSION TITLE
CMAKE gives warning unnecassarily

### DISCUSSION THREAD

**Discussion by BobManMasher** [gerbera/gerbera#1877] [Q&A] [answered]
I have compiled fmtlib-8.0.1 with build_shared_libs=true and spdlog-1.9.2 using the external fmt option however everytime I run CMAKE, I get:

> --Found spdlog: /usr/local/lib/cmake/spdlog/spdlogConfig.cmake (found suitable version "1.9.2", minimum required is "1.8.2")
> CMake Warning at cmake/Findspdlog.cmake:11 (message):
>   spdlog is built without SPDLOG_FMT_EXTERNAL.
> 
>   Since Gerbera uses fmt library internally the spdlog's bundled version wlll
>   be in conflict
> 
>   It is strongly recommended to rebuild spdlog without bundled fmt
> Call Stack (most recent call first):
>   CMakeLists.txt:403 (find_package)

Line 403 is just the find_package for spdlog.

Is this just a warning or is it detecting a problem with the spdlog build that I haven't spotted?

**Comment by whyman** [comment, marked as answer]
Hmm, thats interesting, sounds like a bug, can you paste your  /usr/local/lib/cmake/spdlog/spdlogConfig.cmake?

**Reply by BobManMasher** [reply]
Hi 
I run CMAKE as follows:
'cmake -DSPDLOG_FMT_EXTERNAL=ON'
Note: I did query the syntax with the devs for this and it stopped loading or needing fmtlib7

and this is the content of the spdlogConfig.cmake file:
> cat /usr/local/lib/cmake/spdlog/spdlogConfig.cmake
> # Copyright(c) 2019 spdlog authors
> # Distributed under the MIT License (http://opensource.org/licenses/MIT)
> 
> 
> ####### Expanded from @PACKAGE_INIT@ by configure_package_config_file() #######
> ####### Any changes to this file will be overwritten by the next CMake run ####
> ####### The input file was spdlogConfig.cmake.in                            ########
> 
> get_filename_component(PACKAGE_PREFIX_DIR "${CMAKE_CURRENT_LIST_DIR}/../../../" ABSOLUTE)
> 
> macro(set_and_check _var _file)
>   set(${_var} "${_file}")
>   if(NOT EXISTS "${_file}")
>     message(FATAL_ERROR "File or directory ${_file} referenced by variable ${_var} does not exist !")
>   endif()
> endmacro()
> 
> macro(check_required_components _NAME)
>   foreach(comp ${${_NAME}_FIND_COMPONENTS})
>     if(NOT ${_NAME}_${comp}_FOUND)
>       if(${_NAME}_FIND_REQUIRED_${comp})
>         set(${_NAME}_FOUND FALSE)
>       endif()
>     endif()
>   endforeach()
> endmacro()
> 
> ####################################################################################
> 
> find_package(Threads REQUIRED)
> 
> set(SPDLOG_FMT_EXTERNAL OFF)
> set(config_targets_file spdlogConfigTargets.cmake)
> 
> if(SPDLOG_FMT_EXTERNAL)
>     include(CMakeFindDependencyMacro)
>     find_dependency(fmt CONFIG)
> endif()
> 
> include("${CMAKE_CURRENT_LIST_DIR}/${config_targets_file}")
> 
> check_required_components(spdlog)
> 

BTW - I have finally managed to compile Gerbera 1.9.0 with no errors and warnings apart from this one and am testing it now. Will produce updated instructions later.

**Reply by whyman** [reply]
Notice in your config file it says `set(SPDLOG_FMT_EXTERNAL OFF)`

**Reply by BobManMasher** [reply]
I saw that Ian and was puzzled ...  I actually did a fresh build of fmt, spdlog and gerbera last night (in that order) so that should be the latest file. I will go back to the dev's at spdlog and see what they say.

**Reply by KarlStraussberger** [reply]
Just make sure that no other instance of fmt is already installed. I remember a similar issue some time ago.

**Reply by BobManMasher** [reply]
Been chasing my tail a bit on this but a breakthrough this evening - I deleted /usr/local/lib/cmake/spdlog/spdlogConfig.cmake, rebuilt spdlog and installed the package and rebuilt gerbera.  The cmake no longer gives the warning and just says found spdlog and no mention of the external_fmt option. The build completed Ok and is now running on the test system.  I don't know what made the spdlogConfig.cmake but once it was there, whatever way I built spdlog did not change or update it.
Thanks Ian for your assistance.
